lib/Doctrine/ruleset.xml

```diff
@@ -3,31 +3,180 @@
     <description>The Doctrine coding standard.</description>
 
     <!-- Import PSR-2 coding standard (base) -->
-    <rule ref="PSR2">
-        <!-- Using \Doctrine\Sniffs\Spacing\ControlStructureSniff instead -->
-        <exclude name="PSR2.ControlStructures.ControlStructureSpacing"/>
-    </rule>
+    <rule ref="PSR2"/>
 
-    <!-- Disallow else if in favour elseif -->
-    <rule ref="PSR2.ControlStructures.ElseIfDeclaration.NotAllowed">
-        <type>error</type>
+    <!-- Forbid `array(...)` -->
+    <rule ref="Generic.Arrays.DisallowLongArraySyntax"/>
+    <!-- Forbid duplicate classes -->
+    <rule ref="Generic.Classes.DuplicateClassName"/>
+    <!-- Forbid empty statements -->
+    <rule ref="Generic.CodeAnalysis.EmptyStatement">
+        <!-- But allow empty catch -->
+        <exclude name="Generic.CodeAnalysis.EmptyStatement.DetectedCATCH"/>
     </rule>
-
+    <!-- Forbid final methods in final classes -->
+    <rule ref="Generic.CodeAnalysis.UnnecessaryFinalModifier"/>
+    <!-- Forbid useless empty method overrides -->
+    <rule ref="Generic.CodeAnalysis.UselessOverridingMethod"/>
+    <!-- Forbid inline control structures, i.e. `if ($foo) bar();` -->
+    <rule ref="Generic.ControlStructures.InlineControlStructure"/>
+    <!-- Forbid inline HTML in PHP code -->
+    <rule ref="Generic.Files.InlineHTML"/>
     <!-- Align corresponding assignment statement tokens -->
     <rule ref="Generic.Formatting.MultipleStatementAlignment">
         <properties>
             <property name="error" value="true"/>
         </properties>
     </rule>
-
     <!-- Force whitespace after a type cast -->
-    <rule ref="Generic.Formatting.SpaceAfterCast" />
-
+    <rule ref="Generic.Formatting.SpaceAfterCast"/>
+    <!-- Force whitespace after `!` -->
+    <rule ref="Generic.Formatting.SpaceAfterNot"/>
+    <!-- Forbid PHP 4 constructors -->
+    <rule ref="Generic.NamingConventions.ConstructorName"/>
+    <!-- Forbid any content before opening tag -->
+    <rule ref="Generic.PHP.CharacterBeforePHPOpeningTag"/>
+    <!-- Forbid deprecated functions -->
+    <rule ref="Generic.PHP.DeprecatedFunctions"/>
+    <!-- Forbid alias functions, i.e. `sizeof()`, `delete()` -->
+    <rule ref="Generic.PHP.ForbiddenFunctions"/>
+    <!-- Forbid useless inline string concatenation -->
+    <rule ref="Generic.Strings.UnnecessaryStringConcat">
+        <!-- But multiline is useful for readability -->
+        <properties>
+            <property name="allowMultiline" value="true"/>
+        </properties>
+    </rule>
+    <!-- Force scope changes to increase indentation -->
+    <rule ref="Generic.WhiteSpace.ScopeIndent"/>
+    <!-- Require EOL at EOF -->
+    <rule ref="Generic.Files.EndFileNewline"/>
+    <!-- Forbid multiple classes per file -->
+    <rule ref="Generic.Files.OneClassPerFile"/>
```

Comment:
This one will be problematic for ORM where tests have their entities defined in the same file. Could be excluded on project level though.

Reply:
Should be excluded per-directory IMO

Reply:
If that is possible to configure, even better. 👍

Reply:
It would simply require a custom local `phpcs.xml` file per directory to be inspected, with overrides for disabled rules

Reply:
Isn't this covered by `PSR1.Classes.ClassDeclaration.MultipleClasses` (which is part of PSR-2)? If so, we already exclude things in the ORM

https://github.com/doctrine/doctrine2/blob/374e7ace49d864dad8cddbc55346447c8a6a2083/phpcs.xml.dist#L20-L22

Reply:
Didn't realize it's already included 👍, we can remove these checks then, PSR-1 sniff provides check for classes/interfaces/traits at once.

Reply:
Removed in 2ac121d.